compare_experiments/consistency_common_DEGs.py

```diff
@@ -0,0 +1,212 @@
+    [set(sophie_crow_common), set(sophie_recount2_common)],
+    set_labels=("Array (Crow et al.)", "RNA-seq (recount2)"),
+)
+
+cross_platform_venn.get_patch_by_id("11").set_color("purple")
+cross_platform_venn.get_patch_by_id("11").set_edgecolor("none")
+cross_platform_venn.get_patch_by_id("11").set_alpha(0.3)
+cross_platform_venn.get_patch_by_id("01").set_color("blue")
+cross_platform_venn.get_patch_by_id("01").set_edgecolor("none")
+cross_platform_venn.get_patch_by_id("01").set_alpha(0.3)
+
+plt.title("Same context, different platforms", fontsize=16, fontname="Verdana")
+for text in cross_platform_venn.set_labels:
+    text.set_fontsize(14)
+    text.set_fontname("Verdana")
+
+for text in cross_platform_venn.subset_labels:
+    text.set_fontsize(12)
+    text.set_fontname("Verdana")
+
+# Save figure
+matplotlib.pyplot.savefig(
+    "cross_platform_venn.svg",
+    format="svg",
+    bbox_inches="tight",
+    transparent=True,
+    pad_inches=0,
+    dpi=300,
+)
+
+# +
+cross_context_venn = venn2(
+    [set(sophie_crow_common), set(sophie_powers_common)],
+    set_labels=("Mixed (Crow et al.)", "Cancer (Powers et al.)"),
+)
+
+cross_context_venn.get_patch_by_id("11").set_color("purple")
+cross_context_venn.get_patch_by_id("11").set_edgecolor("none")
+cross_context_venn.get_patch_by_id("11").set_alpha(0.3)
+cross_context_venn.get_patch_by_id("01").set_color("blue")
+cross_context_venn.get_patch_by_id("01").set_edgecolor("none")
+cross_context_venn.get_patch_by_id("01").set_alpha(0.3)
+
+plt.title("Different context, same platforms", fontsize=16, fontname="Verdana")
+
+for text in cross_context_venn.set_labels:
+    text.set_fontsize(14)
+    text.set_fontname("Verdana")
+
+for text in cross_context_venn.subset_labels:
+    text.set_fontsize(12)
+    text.set_fontname("Verdana")
+
+# Save figure
+matplotlib.pyplot.savefig(
+    "cross_context_venn.svg",
+    format="svg",
+    bbox_inches="tight",
+    transparent=True,
+    pad_inches=0,
+    dpi=300,
+)
+
+# +
+cross_context_venn = venn3(
+    [set(sophie_crow_common), set(sophie_powers_common), set(sophie_recount2_common)],
+    set_labels=("Crow et al.", "Powers et al.", "recount2"),
+)
+
+"""cross_context_venn.get_patch_by_id("11").set_color("purple")
+cross_context_venn.get_patch_by_id("11").set_edgecolor("none")
+cross_context_venn.get_patch_by_id("11").set_alpha(0.3)
+cross_context_venn.get_patch_by_id("01").set_color("blue")
+cross_context_venn.get_patch_by_id("01").set_edgecolor("none")
+cross_context_venn.get_patch_by_id("01").set_alpha(0.3)"""
+
+plt.title("Across all compendia")
+
+# Save figure
+matplotlib.pyplot.savefig(
+    "cross_all_venn.svg",
+    format="svg",
+    bbox_inches="tight",
+    transparent=True,
+    pad_inches=0,
+    dpi=300,
+)
+# -
+
+# It looks like the majority of the top 20% commonly changed genes are not consistent across all compendia, but there are some genes that are consistent across platform and context (consistent across 2 compendia but not all 3)
```

Comment:
This is interesting - I wonder how much overlap you'd expect by chance. You'd have to think carefully about your null model though, since probably not all genes are equally likely to be DE in general. Depends on how convinced you think reviewers will be by these plots (the common genes seem to be somewhat consistent but not _super_ consistent/obvious).

Reply:
This is a very good question. The reviewer that mentioned this, seemed accept our approach, but was curious about the boundaries of SOPHIE, specifically how robust common DEGs were across compendia. I think it makes sense that there are some "general common DEGs" that are robust across context and compendia. But there are differences that I would guess are due to the different contexts or technical differences in platform. We tried to do a little digging into the technical differences that found some genes to be common in only the RNA-seq setting. 

I definitely think that its a very good question: How robust are common DEGs across different types of compendia? What accounts for the difference in some genes being common in one compendia vs another? I don't think that this is the paper to really explore those questions. So this analysis as well as our RNA-seq only common DEGs analysis that is another supplemental are peaks into starting to look into that. And I agree that in order to do this more explicitly in the future, we would need to think carefully about a control - one idea i had for a possible negative control is to use specific genes, though perhaps that's circular.

Reply:
Got it - I'm convinced! Hopefully your reviewers feel the same 🙃